9/Grains/src/TimerWorld.cs

```diff
@@ -0,0 +1,101 @@
+
+namespace Grains
+{
+    public class TimerWorld : Grain, ITimerWorld
+    {
+        private readonly ILogger<TimerWorld> _logger;
+        private IDisposable _timer;
+        private int _executionCount;
+
+        public TimerWorld(ILogger<TimerWorld> logger)
+        {
+            _logger = logger;
+        }
+
+        public Task<string> ActivateTimer(GrainCancellationToken grainCancellationToken)
+        {
+            if (grainCancellationToken.CancellationToken.IsCancellationRequested)
+            {
+                return Task.FromResult("Cancellation requested");
+            }
+
+            string result;
+
+            if (_timer == null)
+            {
+                _timer = RegisterTimer(TimerCallback, null, TimeSpan.FromSeconds(2), TimeSpan.FromSeconds(1));
+                result = $"Timer registered - {IdentityString}";
+            }
+            else
+            {
+                result = $"Timer already registered - {IdentityString}";
+            }
+
+            return Task.FromResult(result);
+        }
+
+        public Task<string> DeactivateTimer(GrainCancellationToken grainCancellationToken)
+        {
+            if (grainCancellationToken.CancellationToken.IsCancellationRequested)
+            {
+                return Task.FromResult("Cancellation requested");
+            }
+
+            string result;
+
+            if (_timer == null)
+            {
+                result = $"Timer is not registered - {IdentityString}";
+            }
+            else
+            {
+                _timer.Dispose();
+                _timer = null;
+                result = $"Timer unregistered - {IdentityString}";
+            }
+
+            return Task.FromResult(result);
+        }
+
+        public Task<string> DeactivateGrain(GrainCancellationToken grainCancellationToken)
+        {
+            if (grainCancellationToken.CancellationToken.IsCancellationRequested)
+            {
+                return Task.FromResult("Cancellation requested");
+            }
+
+            string result;
+
+            if (_timer == null)
+            {
+                result = $"Timer is not registered - {IdentityString}";
+            }
+            else
+            {
+                DeactivateOnIdle();
+                result = $"Grain deactivation requested - {IdentityString}";
+            }
+
+            return Task.FromResult(result);
+        }
+
+        private async Task TimerCallback(object state)
+        {
+            _executionCount++;
+
+            var nowInSeconds = DateTime.UtcNow.Second;
+
+            _logger.LogInformation($"{nowInSeconds}: This timer has executed {_executionCount} times - {IdentityString}");
+
+            // this is just to demonstrate that the timer waits for `period`
+            // after the execution finishes before starting the next one
+            // meaning that the log above will happen every 5 seconds
+            await Task.Delay(4000);
+        }
+    }
+}
```

Comment:
Oh, I see what you're doing. This is great and I like the fact that we're registering a timer on grain method execution and not `OnActivateAsync`, but I'm wondering, is it not too much for a simple `Timers` example? The idea I have for this repo is to trim all the fat we don't need to get an example's point across and here we have:

* Web API
* old Web API examples (the whole `hello world` path)
* grain deactivation
* etc.

What I had in mind was something super simple and limited in scope, like an example I wrote sometime earlier last year: https://github.com/PiotrJustyna/orleans-sandbox/tree/main/hello-world-timer

So my thinking is:

* timers are quite easy to set up, so they shouldn't need too much code/configuration around them
* orleans timers aren't actually so trivial to work with and they are not exactly working the same way as the regular timers do:
  * asynchronous callbacks cannot overlap so their executions can delay each other (https://dotnet.github.io/orleans/docs/grains/timers_and_reminders.html#timer-usage). I illustrated it in my example (linked in this comment, not sure if I succeeded getting the point across, though, so feel free to improve it)
  * grains are single-threaded which can lead to interesting observations
  * timer registration actually should not be done at construction time, but rather on activation or grain method execution, which your example partially demonstrates

So in general, I think what would be most helpful for our readers is:

* dramatically less code surrounding the actual demo (your new grain) - what I would suggest is to take e.g. Solution 3 as your starting point and create e.g. Solution 3b.
* really interesting examples illustrating the gotchas of orleans timers as compared to the regular .net timers
* maybe we could use a grain per timer example to keep the examples well isolated - don't want the grain code to get too messy